velox/exec/Window.cpp

```diff
@@ -350,10 +397,20 @@ void Window::callApplyForPartitionRows(
     rawPeerEnds[j] = peerEndRow_ - 1 - firstPartitionRow;
   }
 
-  auto updateFrameBounds = [&](vector_size_t* rawFrameBounds,
-                               core::WindowNode::BoundType boundType,
-                               core::WindowNode::WindowType type,
-                               bool isStartBound) -> void {
+  auto readFrameValues = [&](FrameChannelArg& frameArg) -> void {
+    if (frameArg.index != kConstantChannel) {
+      frameArg.value->resize(numRows);
+      windowPartition_->extractColumn(
+          frameArg.index, partitionOffset_, numRows, 0, frameArg.value);
+    }
+  };
+
+  auto updateFrameBounds =
+      [&](vector_size_t* rawFrameBounds,
```

Comment:
The input parameters appear before output parameters in the argument list. A good order here would be windowType, boundType, isStartBound, frameArg, rawFrameBounds.

Reply:
Thanks for the suggestion but since the frameArg is optional, it is retained as the last parameter. The rawFrameBounds which is the output is moved to the penultimate position.